logwrap/repr_utils.py

```diff
@@ -563,9 +573,17 @@ def _repr_iterable_items(
         """
         next_indent: int = self.next_indent(indent)
         buf: list[str] = []
+        ix = 0
+
         for elem in src:
```

Comment:
Better `for idx, elem in enumerate(src, start=1):` and check if `idx == max_iter`